### DISCUSSION TITLE
Welcome to GeoChemFoam Discussions!

### DISCUSSION THREAD

**Discussion by GeoChemFoam** [GeoChemFoam/GeoChemFoam#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Li-ay** [comment]
I intend to use DBS to simulate mineral dissolution with two different reaction rates. How should I distinguish between different minerals

**Reply by julienmaes** [reply]
Hi Li-ay,

It's not directly possible with GeoChemFoam-5.0, the functionality is still in development and will appear in future release.

The way to do this is to create two different epsilon field and two different reaction object. It's significant development but someone with good knowledge of openFoam should be able to do it.

Regards,

Julien

**Comment by Yeyang28** [comment]
Hey, Professor
I try to install "GeoChemFoam-5.0", but It failed. It complains about missing files. Is there a "ThirdParty/pPhreeqcRm/lib" folder missing in the "ThirdParty/pPhreeqcRm"? Since other "GeoChemFoam-4.X"  has it. Or I have to download phreeqcrm.

Best
Yang

**Reply by julienmaes** [reply]
Allwmake should create these folders and files. What error do you get?

**Reply by Yeyang28** [reply]
> Allwmake should create these folders and files. What error do you get?

I can successfully run GeochemFoam-4.8 without errors under Foam-extend following instructions.
For GeochemFoam-5.0 under Openfoam2212, I tried again, but the "lib" directory in the "PhreeqcRM" is absent or not correctly set up at all.

Best
Yang

**Reply by julienmaes** [reply]
What happen if you do a Allwclean, then go inside ThirdParty/PhreeqcRM and run ./Allwmake from there?

**Reply by Yeyang28** [reply]
> What happen if you do a Allwclean, then go inside ThirdParty/PhreeqcRM and run ./Allwmake from there?

Problem solved! Thank you for your patience, professor. The reason is I should not use Ubuntu 22.04. The new Ubuntu version is not supported. I switch back to Ubuntu 18.04 and the compilation is successful works.  I am excited about your work, hope I can study a lot. 

Thank you!

Best
Yang

**Comment by liaoyang123** [comment]
Hi      Julien
My model comes from the HM12_2_2_4 of GeoChemFoam. I used it in DBS and I was able to run ./createMesh and initCase.sh successfully. But when I run runCase0.sh I get the following error, what should I do?


run reactiveTransportDBSFoam for 1e-06 sec in parallel on 16 processors
[4] #0  [6] #0  [8] #0  [12] #0  [14] #0  Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)[9] Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)[15] Foam::error::printStack(Foam::Ostream&)[1] #0  #0  [7] #0  #0  [3] #0  [11] [13] #0[10] #0    [0] [2] #Foam::error::printStack(Foam::Ostream&)[5] #Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)#0  Foam::error::printStack(Foam::Ostream&)#0  0  0  Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)Foam::error::printStack(Foam::Ostream&)--------------------------------------------------------------------------
A process has executed an operation involving a call to the
"fork()" system call to create a child process.  Open MPI is currently
operating in a condition that could result in memory corruption or
other system errors; your job may hang, crash, or produce silent
data corruption.  The use of fork() (or system() or other calls that
create child processes) is strongly discouraged.

The process that invoked fork was:

  Local host:          

If you are *absolutely sure* that your application will successfully
and correctly survive a call to fork(), you may disable this warning
by setting the mpi_warn_on_fork MCA parameter to 0.

**Comment by liaoyang123** [comment]
Moreover,I think it might have something to do with setting boundary conditions, because I learned a little bit in https://www.cfd-online.com/Forums/openfoam/103931-open-mpi-fork-error.html.

**Comment by liaoyang123** [comment]
Thank you. I plan to use reactiveTransportDBSFoam and try to use this model to simulate dissolution patterns under different Pe and Ki conditions. I've tried refining the grid and this one does work, but the Courant Number get big when I run it for a while. Should I set my injection speed very low?Is there any other better solution?

**Comment by Javad-si** [comment]
Dear Professor Maes,
I hope you are doing good.

I wanted to ask about the compatibility of the GeochemFoam 5 with different Ubuntu versions; as I have tried to compile it on Ubuntu 22.04 and I faced errors leading to an unsuccessful installation. does it only work on Ubuntu 18.04?

Your guidance is greatly appreciated.

**Comment by Kolihanar** [comment]
Excellent solver, looking forward to launching chemical reaction solvers with multiple reactants
